ui/src/main/resources/FileManagerCode/DriveSheet.xml

```diff
@@ -1899,53 +1909,92 @@ require(['jquery', 'angular', 'JobRunner', 'angular-resource', 'angular-ui-boots
      */
-    $(document).on('hidden.bs.modal', '#similarFile', function() {
-      if ($('#similarFile').data('formSubmitted')) {
+    $(document).on('hidden.bs.modal', '#similarFileModal', function() {
+      if ($('#similarFileModal').data('formSubmitted')) {
         var similarFileForm = $(this).find('.modal-body form').serializeArray();
-        var results = {};
-        $.each(similarFileForm, function(i, field) {
-          results[field.name] = field.value;
-        });
-
-        repeatAction = results.repeatAction ? results.uploadAction : null;
-        var replacedFile = $('#similarFile').data('similarFiles').filter(function(file) {
-          return file.id == results.replacedFile;
+        // The form has no fields with multiple values.
+        var uploadAction = getFormFieldValue(similarFileForm, 'uploadAction');
+        var replacedFileId = getFormFieldValue(similarFileForm, 'replacedFile');
+        var repeatAction = getFormFieldValue(similarFileForm, 'repeatAction');
+
+        $('#similarFileModal').data('repeatAction', repeatAction ? uploadAction : null);
+        var replacedFile = $('#similarFileModal').data('similarFiles').filter(function(file) {
+          return file.id == replacedFileId;
         })[0];
 
-        $('#similarFile').data('deferred').resolve(results.uploadAction, replacedFile);
+        $('#similarFileModal').data('deferred').resolve(uploadAction, replacedFile);
       } else {
-        $('#similarFile').data('deferred').resolve(uploadActions.ABORT);
+        $('#similarFileModal').data('deferred').resolve(uploadActions.ABORT);
       }
-
-      // Clear all data attributes of the hidden modal.
-      $.each($('#similarFile').data(), function (id) {
-        $("#similarFile").removeAttr("data-" + id);
-      });
     });
 
     /**
      * Add similar files as an option for the user to choose the replaced file.
      */
-    $(document).on('show.bs.modal', '#similarFile', function (e) {
-      $('#fileName').text($('#similarFile').data('fileName'));
+    $(document).on('show.bs.modal', '#similarFileModal', function (e) {
+      $('#similarFileModal .fileName').text($('#similarFileModal').data('fileName'));
       $('#replacedFile').empty();
-      $.each($('#similarFile').data('similarFiles'), function() {
+      $.each($('#similarFileModal').data('similarFiles'), function() {
         var content = this.name + ' (' + this.size + ')';
-        var option = $('&lt;option&gt;&lt;/option')
+        var option = $('&lt;option&gt;&lt;/option&gt;')
           .attr('value', this.id)
           .attr('title', new XWiki.Document(this.id).getURL())
           .text(content);
         $('#replacedFile').append(option);
       });
     });
 
+    var isSameUpload = function(upload1, upload2, checkReplacedFile) {
+      // Check if each path includes completely the other path.
+      var isSameParent = $(upload1.path).not(upload2.path).length == 0
+        &amp;&amp; $(upload2.path).not(upload1.path).length == 0;
```

Comment:
Are you sure this is right? What type is ``path``? An array of what? Strings or some objects? I tried this in the JavaScript console:

    jQuery(["one", "two"]).not(["two", "one"])
    jQuery(["two", "one"]).not(["one", "two"])

and both have 0 length although ``one/two`` is not the same as ``two/one`` if you consider that the array represents a path (not an unordered set).